### PR TITLE
Feature/selenium3

### PR DESCRIPTION
**Pull request by c3riley**

### CODE REVIEW

File: testah-junit/src/main/java/org/testah/driver/web/browser/FirefoxBrowser.java [hunk 1]
```diff
@@ -1,41 +1,47 @@
...
     /*
      * (non-Javadoc)
      *
      * @see org.testah.driver.web.browser.AbstractBrowser#getDriverBinay()
      */
     @Override
     public FirefoxBrowser getDriverBinay() {
+        FirefoxDriverManager.getInstance().setup();
+        System.setProperty("webdriver.gecko.driver", FirefoxDriverManager.getInstance().getBinaryPath());
+        TS.params().setWebDriver_firefoxDriverBinary(FirefoxDriverManager.getInstance().getBinaryPath());
```
Comment (gweiske) [inline testah-junit/src/main/java/org/testah/driver/web/browser/FirefoxBrowser.java]: setting the binary path twice?

File: testah-junit/src/test/java/org/testah/driver/http/requests/AbstractRequestDtoTest.java [hunk 2]
```diff
@@ -12,12 +13,13 @@
...
 
-        PostRequestDto post = new PostRequestDto(url);
+        final PostRequestDto post = new PostRequestDto(url);
         post.setPayload(payloadString);
         Assert.assertEquals(payloadString, post.getPayloadString());
 
         post.setPayload(new TestAbstractDtoBaseDto().setValue(payloadString));
-        Assert.assertEquals("{\n  \"value\" : \"This is a payload String\"\n}", post.getPayloadString());
+        System.out.println(StringEscapeUtils.escapeJava(post.getPayloadString()));
+        Assert.assertEquals("{\r\n  \"value\" : \"This is a payload String\"\r\n}", post.getPayloadString());
```
Comment (gweiske) [inline testah-junit/src/test/java/org/testah/driver/http/requests/AbstractRequestDtoTest.java]: do the  new line chars add anything to the test? maybe just "contains"?

File: testah-junit/src/main/java/org/testah/driver/web/browser/FirefoxBrowser.java [hunk 3]
```diff
@@ -68,13 +74,14 @@ public DesiredCapabilities createCapabilities() {
         capabilities.setCapability("nativeEvents", false);
         capabilities.setCapability(FirefoxDriver.PROFILE, profile);
         capabilities.setCapability("elementScrollBehavior", 1);
-        if (null != TS.params().getWebDriver_firefoxDriverBinary() && TS.params().getWebDriver_firefoxDriverBinary().length() > 0) {
+        if (null != TS.params().getWebDriver_firefoxDriverBinary()
```
Comment (jrdrew) [inline testah-junit/src/main/java/org/testah/driver/web/browser/FirefoxBrowser.java]: https://commons.apache.org/proper/commons-collections/javadocs/api-3.2.2/org/apache/commons/collections/CollectionUtils.html#isEmpty(java.util.Collection) seems cleaner

File: testah-junit/src/main/java/org/testah/driver/web/browser/GoogleChromeBrowser.java [hunk 4]
```diff
@@ -8,44 +14,44 @@
...
-     * @see org.testah.driver.web.browser.AbstractBrowser#getWebDriver(org.openqa.selenium.remote.DesiredCapabilities)
+     * @see
+     * org.testah.driver.web.browser.AbstractBrowser#getWebDriver(org.openqa.
+     * selenium.remote.DesiredCapabilities)
      */
     public WebDriver getWebDriver(final DesiredCapabilities capabilities) {
         if (null == service) {
             return new ChromeDriver(capabilities);
-        } else {
+        }
```
Comment (jrdrew) [inline testah-junit/src/main/java/org/testah/driver/web/browser/GoogleChromeBrowser.java]: formatting?

File: testah-junit/src/main/java/org/testah/driver/web/browser/FirefoxBrowser.java (hunk 3 above)
Comment (c3riley) [reply testah-junit/src/main/java/org/testah/driver/web/browser/FirefoxBrowser.java]: Think you mean string utils?  Ya much cleaner, will update

**Comment by c3riley** [comment]
merging into gcwdev, this is hard versioned when it comes it, since its getting selenium 3, it will no longer work with older firefox 47 or older.

**Pull request merged**